plugins/woocommerce/includes/wc-order-functions.php

```diff
@@ -1148,5 +1148,18 @@ function wc_create_order_note( $order_id, $note, $is_customer_note = false, $add
  * @return bool         True on success, false on failure.
  */
 function wc_delete_order_note( $note_id ) {
-	return wp_delete_comment( $note_id, true );
+	$delete = wp_delete_comment( $note_id, true );
+
+	if($delete) {
+		/**
+		 * Action hook fired after an order note is deleted.
+		 *
+		 * @param int      $note_id	Order note ID.
+		 *
+		 * @since 7.7.0
+		 */
+		do_action( 'woocommerce_order_note_deleted', $note_id);
```

Comment:
If we are executing an action, let's fetch the `$note` object, and pass it as a parameter. Just the note_id which is deleted is probably not very useful, especially in systems which already don't have another copy of the note identified by the ID.

Reply:
I agree, but could you help me with that, how do i fetch the `$note` object? and what about `deleted_comment`? of course order notes are WordPress comments but I just need it to be more specific.

Reply:
@evokelektrique I apologize for the delay in follow up. Use `$note = wc_get_order_note( $note_id );` before line 1151.

Reply:
I'll get into it